snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -605,9 +610,40 @@ def _wrap_apps(self, apps: Dict[str, Any]) -> Dict[str, Any]:
             adapter = Adapter[app.pop("adapter").upper()]
             if adapter == Adapter.LEGACY:
                 self._wrap_app(app_name, app)
+            elif adapter == Adapter.FULL:
+                self._add_command_chain_to_app(app_name, app)
             self._generate_desktop_file(app_name, app)
         return apps
 
+    def _add_command_chain_to_app(self, app_name, app):
+        # First, validate the commands. Without a wrapper, we're bound by snapd's
+        # restrictions. No quotes, and the first string must be a relative path from
+        # the root of the snap.
+        commands = (app[k] for k in ("command", "stop-command") if k in app)
+        for command in commands:
+            if not _APP_COMMAND_PATTERN.match(command):
+                raise errors.InvalidAppCommandFormatError(command, app_name)
+
+            command_without_args = command.split()[0]
```

Comment:
Do you want to check that this does not start with slash for a fine grained error message? The regex does not capture that.

Reply:
Good catch-- actually no, we don't want to error, it seems snapd supports that, so we must as well. I've fixed the check and added a test to ensure this works.

Reply:
We chatted about this in IRC, and decided that supporting this would end up confusing people, so I've pushed an update that disallows leading slashes.